overlord/overlord.go

```diff
@@ -209,7 +215,45 @@ func (o *Overlord) addManager(mgr StateManager) {
 	o.stateEng.AddManager(mgr)
 }
 
-func loadState(backend state.Backend, restartHandler restart.Handler) (*state.State, error) {
+func initStateFileLock() (*osutil.FileLock, error) {
+	lockFilePath := dirs.SnapStateLockFile
+	if err := os.MkdirAll(filepath.Dir(lockFilePath), 0755); err != nil {
+		return nil, err
+	}
+
+	return osutil.NewFileLockWithMode(lockFilePath, 0644)
+}
+
+func lockWithTimeout(l *osutil.FileLock, timeout time.Duration) error {
+	startTime := time.Now()
+	const retryInterval = time.Second
+	for {
+		err := l.TryLock()
+		if err != osutil.ErrAlreadyLocked {
+			return err
+		}
+
```

Comment:
if  we don't get the lock on the first attempt the retry wait is all unexpected startup time,  I think we should sd-notify EXTEND_TIMEOUT_USEC=timeout  systemd so that we don't get killed prematurely later. We had issues of that kind previously.

Reply:
if timeout isn't too big (<< 1min) another option is to just remember how much we waited and add it to the result of StartupTimeout which daemon uses already to pass along as EXTEND_TIMEOUT_USEC

Reply:
Done